automatic/youtube-dl/update.ps1

```diff
@@ -27,22 +24,37 @@ function global:au_SearchReplace {
 }
 
 function global:au_GetLatest {
-  $download_page = Invoke-WebRequest -Uri $releases -UseBasicParsing
-
-  $re    = '\.exe$'
-  $url   = $domain + "/" + ($download_page.Links | ? href -match $re | select -First 1 -expand href)
-  $filename = $url.Substring($url.LastIndexOf('/') + 1)
-  ($download_page.Content -match "\(v(2[0-9]{3}[.].*)\)")
-  $version  = $Matches[1]
-  
-  #Update checksum automatically from self-contained executable
-  $checksum = Get-RemoteChecksum -Algorithm "sha512" $url
-  
-  return @{
-    Version = $version
-    URL32 = $url
-    Checksum32 = $checksum
-    ChecksumType32 = "sha512"
+  $streams = [ordered]@{}
+  foreach ($Repository in "youtube-dl","ytdl-nightly") {
+    $LatestRelease = Get-GitHubRelease -Owner ytdl-org -Name $Repository
+    $Url = $LatestRelease.assets.Where{$_.name -eq 'youtube-dl.exe'}.browser_download_url
+
+    $Version = $LatestRelease.tag_name.TrimStart('v')
+    if ($Repository.EndsWith('nightly')) {
+      $Version += '-nightly'
+    }
+
+    $Checksum = if ($ChecksumFile = $LatestRelease.assets.Where{$_.name -eq 'SHA2-512SUMS'}) {
+      $Checksum = (-join [char[]](Invoke-WebRequest -Uri $ChecksumFile.browser_download_url -UseBasicParsing).content).Split("`n") -match "^(?<checksum>.+)(?=\s+youtube-dl\.exe)"
+      $Checksum.Split(' ')[0]
+    } else {
+      Get-RemoteChecksum -Algorithm "sha512" $Url
+    }
```

Comment:
Could this be simplified. Have an assignment with an if statement assigning to the results of a where is overly complicated. 

I _don't think_ we do cuddled elses.

Reply:
```suggestion
    $Checksum = if ($ChecksumFile = $LatestRelease.assets.Where{$_.name -eq 'SHA2-512SUMS'}) {
      ((-join [char[]](Invoke-WebRequest -Uri $ChecksumFile.browser_download_url -UseBasicParsing).content).Split("`n") -match "^(?<checksum>.+)(?=\s+youtube-dl\.exe)").Split(' ')[0]
    } else {
      Get-RemoteChecksum -Algorithm "sha512" $Url
    }
```

Reply:
We don't use cuddled else's as far as I am aware but I'll defer to @AdmiringWorm on that.

With regards to the [complexity of the code](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md#117-clean-code):

> Remember, code is written for humans, not for computers. Otherwise, we'd all write assembly. So it's better to be able to reason about code than it is to take shortcuts that make code harder to decipher.

I know that complexity is subjective, but this needs to be simplified in my opinion. Breaking it out into separate steps would seem to be a good start rather than doing everything in one line. With comments.

Reply:
Throughout different changes, there have been statements that were both cuddled and not cuddled, as well as on its own line and not.
It looks to be more or less an equal amount of these (this mix is due to, historically, the look of the code did not matter, just that it works).

Personally, I would prefer else statements starting on their line (but I am not tied to that).

Also, I agree with the complexity. It makes reading what you are doing hard and needs to be simplified.

Reply:
Broken it out into many lines with comments in my latest push.

I'm happy to argue my case for us adopting cuddled-elses and no-new-line as our overall preference for a few reasons:
- It reduces the potential for contributors accidentally breaking chained statements by inserting new lines between an `if {}` and an `else {}` - to me it offers a very clear linkage.
- I love being able to paste chunks of code into terminals without modification when I'm debugging scripts, which doesn't happen reliably if the `if {}` has evaluated already (and then it makes a complete hash of the else-scriptblock)

Overly Simplistic Example:
![image](https://github.com/chocolatey-community/chocolatey-packages/assets/1975761/0802ea12-5d2d-4184-a17c-e48b34ff88b5)

If I wasn't on holiday for it, I'd say we could talk about it at the upcoming maintainers meeting. If we do so, perhaps we can look to add a rule to the formatting PSD1 and make a formatting commit to standardise once and for all.